deepinv/physics/generator/mri.py

```diff
@@ -1,83 +1,244 @@
+
+    Generate a mask of vertical lines for MRI acceleration with fixed sampling in low frequencies (center of k-space) and Gaussian undersampling in the high frequencies.
+
+    The high frequences are selected according to a tail-adjusted Gaussian pdf. This ensures that the expected number of rows selected is equal to (N / acceleration).
+
+    Supports k-t sampling, where the Gaussian mask varies randomly across time.
+
+    The mask is repeated across channels and randomly varies across batch dimension.
+
+    Algorithm taken from Schlemper et al. `A Deep Cascade of Convolutional Neural Networks for Dynamic MR Image Reconstruction <https://github.com/js3611/Deep-MRI-Reconstruction/blob/master/utils/compressed_sensing.py>`_.
+
+    For parameter descriptions see :class:`deepinv.physics.generator.mri.BaseMaskGenerator`
+
+    Example:
+
+    ::
+
+        generator = GaussianMaskGenerator((2, 8, 64, 64), acceleration=8, center_fraction=0.04) # C, T, H, W
+        params = generator.step(batch_size=1)
+        mask = params["mask"]
+        mask.shape # 1, 2, 8, 64, 64
+
+    """
+
+    def get_pdf(self) -> torch.Tensor:
+        """Create one-dimensional Gaussian probability density function across columns, ignoring any fixed sampling columns.
+
+        Add uniform distribution so that the probability of sampling high-frequency lines is non-zero.
+
+        :return torch.Tensor: unnormalised 1D vector representing pdf evaluated across mask columns.
+        """
+        normal_pdf = lambda length, sensitivity: np.exp(
+            -sensitivity * (np.arange(length) - length / 2) ** 2
         )
-        mask[:, :, center_line_indices] = 1
+        pdf = normal_pdf(self.W, 0.5 / (self.W / 10.0) ** 2)
+        lmda = self.W / (2.0 * self.acc)
 
-        for i in range(batch_size):
-            random_line_indices = np.random.choice(
-                H, size=(num_lines_side // 2,), replace=False
-            )
-            mask[i, :, random_line_indices] = 1
+        pdf += lmda * 1.0 / self.W
+
+        return pdf
+
+
+class UniformMaskGenerator(BaseMaskGenerator):
+    """Generator for MRI Cartesian acceleration masks using uniform (equispaced) non-random undersampling.
+
+    Generate a mask of vertical lines for MRI acceleration with fixed sampling in low frequencies (center of k-space) and equispaced undersampling in the high frequencies.
+
+    The number of lines selected with equal spacing are at a proportion that reaches the desired acceleration rate taking into consideration the number of low-freq lines, so that the total number of lines is (N / acceleration).
+
+    Supports k-t sampling, where the uniform mask is sheared across time.
 
-        return {"mask": torch.cat([mask.float().unsqueeze(1)] * C, dim=1)}
+    The mask is repeated across channels and randomly varies across batch dimension. Based off fastMRI code https://github.com/facebookresearch/fastMRI
```

Comment:
it is a bit confusing that the mask has 'non-random undersampling' but 'randomly varies across batch dimension'?